packages/lesswrong/components/users/DialogueSuggestionsPage.tsx

```diff
@@ -0,0 +1,782 @@
+    backgroundColor: theme.palette.background.paper,
+    color: theme.palette.link.unmarked,
+    whiteSpace: 'nowrap'
+  },
+  newDialogueButton: {
+    maxHeight: `17px`,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    backgroundColor: theme.palette.primary.light,
+    color: 'white',
+    whiteSpace: 'nowrap'
+  },
+  link: {
+    color: theme.palette.primary.main,
+    cursor: 'pointer',
+    '&:hover': {
+      color: theme.palette.primary.light,
+    }
+  },
+  rootFlex: {
+    display: 'flex'
+  },
+  gradientBigTextContainer: {
+    position: 'relative',
+    height: '70px', 
+    overflow: 'auto',
+    color: 'grey', 
+    fontSize: '14px',
+    lineHeight: '1.15em',
+    WebkitMaskImage: 'linear-gradient(to bottom, transparent 0%, black 20%, black 80%, transparent 100%)',
+    '&.scrolled-to-bottom': {
+      WebkitMaskImage: 'linear-gradient(to bottom, transparent 0%, black 20%, black 100%)',
+    },
+    '&.scrolled-to-top': {
+      WebkitMaskImage: 'linear-gradient(to bottom, black 0%, black 80%, transparent 100%)',
+    }
+  },
+  privacyNote: {
+    color: 'grey',
+    fontSize: '1rem',
+  },
+  checkbox: {
+    color: 'default',
+    '&$checked': {
+      color: 'default',
+    },
+  },
+  checked: {},
+  checkboxCheckedMatched: {
+    color: 'green',
+    '&$checked': {
+      color: 'green',
+    },
+  },
+  checkboxCheckedNotMatched: {
+    color: '#ADD8E6',
+    '&$checked': {
+      color: '#00000038',
+    },
+  },
+  centeredText: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+});
+
+const useScrollGradient = (ref: React.RefObject<HTMLDivElement>) => {
+  const [isScrolledToTop, setIsScrolledToTop] = useState(true);
+  const [isScrolledToBottom, setIsScrolledToBottom] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const element = ref.current;
+      if (element) {
+        const atTop = element.scrollTop <= (element.scrollHeight * 0.10);
+        const atBottom = (element.scrollHeight - element.scrollTop) <= (element.clientHeight * 1.10);
+        setIsScrolledToTop(atTop);
+        setIsScrolledToBottom(atBottom);
+      }
+    };
+
+    ref.current?.addEventListener('scroll', handleScroll);
+    return () => ref.current?.removeEventListener('scroll', handleScroll);
+  }, [ref]);
+
+  return { isScrolledToTop, isScrolledToBottom };
+};
+
+const UserBio = ({ classes, userId }: { classes: ClassesType, userId: string }) => {
+  const { document: userData, loading } = useSingle({
+    documentId: userId,
+    collectionName: "Users",
+    fragmentName: "UsersProfile"
+  });
+
+  const bioContainerRef = useRef<HTMLDivElement | null>(null);
+  const { isScrolledToTop, isScrolledToBottom } = useScrollGradient(bioContainerRef);
+
+  return (
+    <div className={`${classes.gradientBigTextContainer} ${isScrolledToTop ? 'scrolled-to-top' : ''} ${isScrolledToBottom ? 'scrolled-to-bottom' : ''}`} ref={bioContainerRef}>
```

Comment:
This should be handled with `classNames`, see other usages for conditional application of class names based on boolean flags